File: AlRasMarketBackend/DataLayer/Models/Category.cs
namespace DataLayer.Models;

public class Category
{
    public byte CategoryId { get; set; }
    public string NameEn { get; set; } = string.Empty;
    public string NameAr { get; set; } = string.Empty;
    public string ImgPath { get; set; } = string.Empty;
    public decimal CommissionPercent { get; set; }
    /// <summary>When true, category is hidden from the mobile app public APIs.</summary>
    public bool IsHide { get; set; }

    public ICollection<Product> Products { get; set; } = new List<Product>();
}

File: AdminDashboard/src/types/category.ts
export type Category = {
  categoryId: number
  nameEn: string
  nameAr: string
  imgPath: string
  commissionPercent: number
  /** When true, category is hidden from the mobile app. */
  isHide: boolean
}

export type CategoriesResponse = {
  count: number
  items: Category[]
}
